### ISSUE TITLE
5.4 Cannot read property 'enumerable' of undefined

### ISSUE THREAD

**Issue by timothyallan** [mobxjs/mobx-utils#191]
Hey all, ViewModel is 5.4 is lighting up my console with waaaay too much red text! 

My class objects which I'm using as ViewModels have the usual observable properties, but occasional functions as well. 

It seems like this is the culprit https://github.com/mobxjs/mobx-utils/blob/5d147c1a43376cb6da71d21197ed5ad21ccf77d5/src/create-view-model.ts#L59

In this loop, https://github.com/mobxjs/mobx-utils/blob/5d147c1a43376cb6da71d21197ed5ad21ccf77d5/src/create-view-model.ts#L46

As soon as the `key` is the name of a function in my class object, I get `Uncaught TypeError: Cannot read property 'enumerable' of undefined` 

and everything explodes. 5.2 works great.

**Comment by ItamarShDev** [comment]
https://github.com/mobxjs/mobx-utils/commit/9aebd222f1e72ae8648f7327184590852962f634

**Comment by ItamarShDev** [comment]
@timothyallan can you please provide a [codesandbox](https://codesandbox.io/s/v3v0my2370) example?

**Comment by timothyallan** [comment]
Sure, it won't let me save it, but simply paste this into the index.js and it crashes.

```
import React from "react";
import { render } from "react-dom";
import { observable, action } from "mobx";
import { observer } from "mobx-react";
import DevTools from "mobx-react-devtools";
import { createViewModel } from 'mobx-utils';

class MyClass {
  @observable firstname;
  @observable lastname;

  someFunction(someParam) {

  }
}

class AppState {
  constructor() {
    const myClass = new MyClass();
    const observableMyClass = createViewModel(myClass);
  }

  @action.bound
  reset() {
    this.timer = 0;
  }
}

const TimerView = observer(({ appState }) => (
  <button onClick={appState.reset}>Seconds passed: {appState.timer}</button>
));


render(
  <div>
    <TimerView appState={new AppState()} />
    <DevTools />
  </div>,
  document.getElementById("root")
);
```

**Comment by ItamarShDev** [comment]
Using arrow function seems to work, like `someFunction = (someParam) => {}`
I am not saying its what should happen, but give my take

your thoughts @mweststrate ?

**Comment by beckend** [comment]
https://github.com/mobxjs/mobx-utils/pull/193

**Comment by mweststrate** [comment]
the arrow function "fixes" it as that makes the method a local (own) property, which is present indeed.

I think a simple fix (strictly speaking not 100% correct) would be:

```
const descriptor = Object.getOwnPropertyDescriptor(model, key)
const enumerable = descriptor ? !!descriptor.enumerable : false
```

**Comment by dr0p** [comment]
Please review this fix https://github.com/mobxjs/mobx-utils/pull/198
Sorry for the miss..

**Comment by timothyallan** [comment]
This seems fixed now in 5.4.1. Thanks!
